pkg/apis/pipeline/register.go

```diff
@@ -41,3 +43,42 @@ const (
 	// ConditionCheckKey is used as the label identifier for a ConditionCheck
 	ConditionCheckKey = "/conditionCheck"
 )
+
+var (
+	// TaskResource represents a Tekton Task
+	TaskResource = schema.GroupResource{
+		Group:    GroupName,
+		Resource: "tasks",
+	}
+	// ClusterTaskResource represents a Tekton ClusterTask
+	ClusterTaskResource = schema.GroupResource{
+		Group:    GroupName,
+		Resource: "clustertasks",
+	}
+	// TaskRunResource represents a Tekton TaskRun
+	TaskRunResource = schema.GroupResource{
+		Group:    GroupName,
+		Resource: "taskruns",
+	}
+	// PipelineResource represents a Tekton Pipeline
+	PipelineResource = schema.GroupResource{
+		Group:    GroupName,
+		Resource: "pipelines",
+	}
+	// PipelineRunResource represents a Tekton PipelineRun
+	PipelineRunResource = schema.GroupResource{
+		Group:    GroupName,
+		Resource: "pipelineruns",
+	}
+
+	// PipelineResourceResource represents a Tekton PipelineResource
+	PipelineResourceResource = schema.GroupResource{
+		Group:    GroupName,
+		Resource: "pipelineresources",
+	}
+	// ConditionResource represents a Tekton Condition
+	ConditionResource = schema.GroupResource{
+		Group:    GroupName,
+		Resource: "conditions",
+	}
+)
```

Comment:
im not 100% clear on why these are required now or where they are used? does the conversion webhook automatically detect these?

Reply:
We do here : https://github.com/tektoncd/pipeline/pull/2413/files#diff-58de452513b7e8d8d3cfea23eb4ae6a8R170 (and we might do at other parts of the code). It makes it a bit easier to get `conditions.tekton.dev` or `pipelineruns.tekton.dev` in code without hard coding those.